sycl/test-e2e/ESIMD/local_accessor_copy_to_from.cpp

```diff
@@ -0,0 +1,135 @@
+//==-- local_accessor_copy_to_from.cpp  - DPC++ ESIMD on-device test -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+//
+// TODO: Enable the test when GPU driver is ready/fixed.
+// XFAIL: opencl || windows || gpu-intel-pvc || gpu-intel-gen12
+// TODO: add support for local_accessors to esimd_emulator.
+// UNSUPPORTED: esimd_emulator
+// The test checks functionality of the gather/scatter local
+// accessor-based ESIMD intrinsics.
+
+#include "esimd_test_utils.hpp"
+
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+template <typename T, unsigned VL> bool test(queue q) {
+  constexpr size_t size = VL;
+
+  std::cout << "Testing T=" << typeid(T).name() << " VL=" << VL << "...\n";
+
+  // The test is going to use size elements of T type.
+  auto Dev = q.get_device();
+  auto DeviceSLMSize = Dev.get_info<sycl::info::device::local_mem_size>();
+  if (DeviceSLMSize < size) {
+    // Report an error - the test needs a fix.
+    std::cerr << "Error: Test needs more SLM memory than device has!"
+              << std::endl;
+    return false;
+  }
+
+  T *A = new T[size];
+
+  for (unsigned i = 0; i < size; ++i) {
+    A[i] = static_cast<T>(i);
+  }
+
+  try {
+    buffer<T, 1> buf(A, range<1>(size));
+    nd_range<1> glob_range{range<1>{1}, range<1>{1}};
```

Comment:
Please make more than 1 work-items per work-group to truly test that write are done to SLM.